### PR TITLE
Edit functionality

### PR DESCRIPTION
**Pull request by moldovanAndrei99**

### CODE REVIEW

File: app/src/main/java/com/example/notepad/viewModel/RecyclerAdapter.kt [hunk 1]
```diff
@@ -28,24 +30,31 @@ class RecyclerAdapter : ListAdapter<Note, RecyclerAdapter.ViewHolder>(NotesCompa
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.itemView.setOnTouchListener { v, event ->
```
Comment (MirelaBocsa) [inline app/src/main/java/com/example/notepad/viewModel/RecyclerAdapter.kt:33]: You can also use **setOnClickListener**

File: app/src/main/java/com/example/notepad/uiControllers/AddEditScreen.kt [hunk 2]
```diff
@@ -15,20 +15,39 @@ import com.example.notepad.viewModel.IAddEditViewModel
...
+            noteId = it.getInt("noteId")
+        }
+
+        noteAddEditViewModel.listener = {
+            binding.editTextTitle.setText(it.title)
+            binding.editTextDescription.setText(it.description)
+        }
+
+        if (!addAction) {
+            noteAddEditViewModel.findNoteById(noteId)
```
Comment (MirelaBocsa) [inline app/src/main/java/com/example/notepad/uiControllers/AddEditScreen.kt:48]: It is better to sent the note through the bundle and not make a query to retrieve it into this screen

File: app/src/main/java/com/example/notepad/uiControllers/AddEditScreen.kt [hunk 3]
```diff
@@ -39,7 +58,11 @@ class AddEditScreen: AppCompatActivity() {
         binding.saveButton.setOnClickListener {
             val title: String = binding.editTextTitle.text.toString()
             val description: String = binding.editTextDescription.text.toString()
-            noteViewModel.insertNote(title, description)
+            if (addAction) {
```
Comment (MirelaBocsa) [inline app/src/main/java/com/example/notepad/uiControllers/AddEditScreen.kt:61]: You can send the entire note object through bundle (or null for add action) and here you can check if the note is null and perform the add

File: app/src/main/java/com/example/notepad/viewModel/AddEditViewModel.kt [hunk 4]
```diff
@@ -16,6 +16,8 @@ class AddEditViewModel(
 ) : ViewModel(), IAddEditViewModel {
     override val update by lazy { MutableLiveData<Boolean>() }
 
+    override var listener: ((note: Note) -> Unit)? = null
```
Comment (MirelaBocsa) [inline app/src/main/java/com/example/notepad/viewModel/AddEditViewModel.kt:19]: LiveData can be used instead

**Review by MirelaBocsa — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/notepad/viewModel/AddEditViewModel.kt (hunk 4 above)
Comment (moldovanAndrei99) [reply app/src/main/java/com/example/notepad/viewModel/AddEditViewModel.kt]: There is no longer need for listener nor LiveData, the text is loaded directly from the note sent in the bundle.

**Review by MirelaBocsa — APPROVED** [review]

**Pull request merged**